crow/recipes/binned_exact.py

```diff
@@ -175,6 +192,37 @@ def function_mapper(
 
         return function_mapper
 
+    def _get_function_to_integrate_impure_counts(
+        self,
+        prediction: Callable[
+            [
+                npt.NDArray[np.float64],
+                npt.NDArray[np.float64],
+                npt.NDArray[np.float64],
+                float,
+            ],
+            npt.NDArray[np.float64],
+        ],
+    ) -> Callable[[npt.NDArray, npt.NDArray], npt.NDArray]:
+        """Returns a callable function that can be evaluated by an integrator.
+
+        This function is responsible for mapping arguments from the numerical integrator
+        to the arguments of the theoretical prediction function.
+        """
+
+        def function_mapper(
+            int_args: npt.NDArray, extra_args: npt.NDArray
+        ) -> npt.NDArray[np.float64]:
+            mass = int_args[:, 0]
+            z = int_args[:, 1]
+            mass_proxy = int_args[:, 2]
```

Comment:
Then here you check again if there is purity or not. If there is,`mass_proxy` is like this, an int arg. If there isnt, mass proxy is an extra arg, but continue to be an NDArray